### ISSUE TITLE
+ Possible issues

### ISSUE THREAD

**Issue by renatocron** [FozzieHi/WorkersSnowflake#1]
Hello, 
First, thank you for posting this code, it really helped me to check how can to do this on javascript.

But after researching a bit more how cloudflare workers instances are recycled, I found that more than one instance can be running on the same colo at the same time (that's issue number 1) and I don't know if you were able to if the wait is working properly on workers, but the Date().getTime() only changes when IO occur, so it's stable during the same request, so the while is wasting cpu and potentially never finish, and the request may fail to finish.

Until Durable Objects API leave closed-beta, I think that generating non-random ids is not possible on workers.

For now, I'm switching back to using two or three services on distinct servers, to run the sequencing and sign it using jwt, then validate the key on worker.

**Comment by FozzieHi** [comment]
@renatocron I agree, this was only meant to be used in smaller environments where the risk of collisions is inherently low due to only one instance running at one co-location at the same time, when Durable Objects is freely available I'm fairly sure it's possible to reduce the chance of collisions to 0.
